storage/gcloud/aio/storage/storage.py

```diff
@@ -115,9 +126,113 @@ def __init__(self, project, service_file, token=None, session=None):
+                'X-Upload-Content-Length': headers['Content-Length']
+            })
+
+            resp = await session.post(url, headers=post_headers, params=params, data=metadata_str, timeout=5)
+            resp.raise_for_status()
+
+            header_with_session_URI = dict(resp.headers)
+            session_URI = header_with_session_URI.get('Location')
+            if not session_URI:
+                raise RuntimeError('Not able to initiate resumable upload session -- no URI in response header')
+
+            return session_URI
+
+        async def do_upload(session_URI: str, data, timeout: int,
+                            headers: dict, session: aiohttp.ClientSession,
+                            retries=5, retry_sleep=1.):
+            tries = 0
+            while True:
+                resp = await session.put(session_URI, headers=headers, data=data, timeout=timeout)
+                tries += 1
+                if resp.status == 200:
+                    break
+                else:
+                    headers.update({'Content-Range': '*/*'})
+                    asyncio.sleep(retry_sleep)
+                if tries > retries:
+                    resp.raise_for_status()
+                    break
+
+            upload_response = await resp.json()
+
+            return upload_response
+
+        session_URI = await initiate_upload_session(object_name, headers, session)
+        upload_response = await do_upload(session_URI, data, timeout, headers, session)
+
+        return upload_response
+
+    async def _download(self, bucket, object_name, params=None, session=None):
+
+        def get_content_type_and_encoding(response_header: dict):
+            content_type_and_encoding = response_header.get('Content-Type')
+            content_type_and_encoding_split = content_type_and_encoding.split(';')
+            content_type = content_type_and_encoding_split[0].lower().strip()
+
+            encoding = None
+            if len(content_type_and_encoding_split) > 1:
+                encoding_str = content_type_and_encoding_split[1].lower().strip()
+                encoding = encoding_str.split('=')[-1]
+
+            return (content_type, encoding)
+
+        token = await self.token.get()
+        # https://cloud.google.com/storage/docs/json_api/#encoding
+        encoded_object_name = quote(object_name, safe='')
+        url = f'{STORAGE_API_ROOT}/{bucket}/o/{encoded_object_name}'
+        headers = {
+            'Authorization': f'Bearer {token}',
+        }
+
+        if not self.session:
+            self.session = aiohttp.ClientSession(conn_timeout=10,
+                                                 read_timeout=10)
+        session = session or self.session
+        response = await session.get(url, headers=headers, params=params or {},
+                                     timeout=60)
+        response.raise_for_status()
+        response_header = dict(response.headers)
```

Comment:
I think `response.headers` is already a `dict`-like object, so this cast is unnecessary.